packages/cli/src/commands/deviceprofiles/update.ts

```diff
@@ -49,7 +49,7 @@ export default class DeviceProfileUpdateCommand extends SelectingInputOutputAPIC
 // Cleanup is done so that the result of a device profile get can be modified and
 // used in an update operation without having to delete the status, owner, and
 // component name fields, which aren't accepted in the update API call.
-export function cleanupRequest(deviceProfileRequest: Partial<DeviceProfile>): DeviceProfileRequest {
+export function cleanupRequest(deviceProfileRequest: Partial<DeviceProfile & { restrictions: unknown }>): DeviceProfileRequest {
```

Comment:
Should we create an issue to update the type for this in the SDK?

Reply:
Good question...it's not in the documentation that this field should ever be there but it must be or @bflorian wouldn't have had to clear it here. (I'm guessing it's a computed field.) What do you think Bob?

Reply:
The type def in the core API is correct. It field isn't there. What I wanted to accomplish is to allow the result of a device profile get call to be used in a create call to make a copy of the profile, without having to go through and remove a bunch of fields. So the proper thing would have been to define a new interface local to the CLI that extends the one from the core API and adds those fields as optional

Reply:
It seems like the API _should_ have a separate type for return values since it's not exactly the same as what is accepted. (Then, we would have a separate type in the core SDK as well.)

We _could_ include the extra type even though the API doesn't but I don't love that idea.

Reply:
> It seems like the API should have a separate type for return values since it's not exactly the same as what is accepted
From what I see it does. We are just trying to avoid having 2 interfaces here for both request and response between endpoints.

There a multiple ways to handle this in clients, one being some sort of adaptation between them as we did here, but this seems convoluted/confusing at first glance. I recommend just creating a ticket with these details here and we can circle back to address.

Reply:
We do already have separate objects for the request and response (`DeviceProfileRequest` and `DeviceProfile` respectively)...and they match the API documentation which does not include the `restrictions` field in either one:

https://smartthings.developer.samsung.com/docs/api-ref/st-api.html#operation/createDeviceProfile

(I don't see it in the response for any of the endpoints--not even in the examples.)